From 1eeef2465914e8c36b1063fe7070727e3abcdb51 Mon Sep 17 00:00:00 2001
From: Tsukasa Kawaguchi <tsukasa.kawaguchi.aw@hitachi.com>
Date: Tue, 13 Jul 2021 15:36:41 +0900
Subject: [PATCH] mk/gcc.mk: Change the path to the library

Change the path to the library if PKG_CONFIG_SYSROOT_DIR is enabled
in the Yocto build error fix.

Upstream-Status: Inappropriate [configuration]

Signed-off-by: Tsukasa Kawaguchi <tsukasa.kawaguchi.aw@hitachi.com>
Signed-off-by: Yoshifumi Hosoya <yoshifumi.hosoya.wj@renesas.com>
---
 mk/gcc.mk | 10 +++++++---
 1 file changed, 7 insertions(+), 3 deletions(-)

diff --git a/mk/gcc.mk b/mk/gcc.mk
index adc77a24f..e871699e2 100644
--- a/mk/gcc.mk
+++ b/mk/gcc.mk
@@ -1,8 +1,12 @@
 
-CC$(sm)		:= $(CROSS_COMPILE_$(sm))gcc
-CXX$(sm)	:= $(CROSS_COMPILE_$(sm))g++
+ifdef PKG_CONFIG_SYSROOT_DIR
+CC$(sm)        := $(CROSS_COMPILE_$(sm))gcc --sysroot=$(PKG_CONFIG_SYSROOT_DIR)
+LD$(sm)        := $(CROSS_COMPILE_$(sm))ld.bfd --sysroot=$(PKG_CONFIG_SYSROOT_DIR)
+else
+CC$(sm)        := $(CROSS_COMPILE_$(sm))gcc
+LD$(sm)        := $(CROSS_COMPILE_$(sm))ld.bfd
+endif
 CPP$(sm)	:= $(CROSS_COMPILE_$(sm))cpp
-LD$(sm)		:= $(CROSS_COMPILE_$(sm))ld.bfd
 AR$(sm)		:= $(CROSS_COMPILE_$(sm))ar
 NM$(sm)		:= $(CROSS_COMPILE_$(sm))nm
 OBJCOPY$(sm)	:= $(CROSS_COMPILE_$(sm))objcopy
-- 
2.26.2.windows.1

